subsystem/kubeapi_test.go

```diff
@@ -520,13 +520,103 @@ var _ = Describe("[kube-api]cluster installation", func() {
+			Namespace: Options.Namespace,
+			Name:      configSecretRef.Name,
+		}
+		configSecret := getSecret(ctx, kubeClient, configkey)
+		Expect(configSecret.Data["kubeconfig"]).NotTo(BeNil())
+	})
+
+	It("None SNO deploy clusterDeployment full install and validate MetaData", func() {
+		By("Create cluster")
+		secretRef := deployLocalObjectSecretIfNeeded(ctx, kubeClient)
+		spec := getDefaultClusterDeploymentSpec(secretRef)
+		deployClusterDeploymentCRD(ctx, kubeClient, spec)
+		clusterKey := types.NamespacedName{
+			Namespace: Options.Namespace,
+			Name:      spec.ClusterName,
+		}
+		cluster := getClusterFromDB(ctx, kubeClient, db, clusterKey, waitForReconcileTimeout)
+		hosts := make([]*models.Host, 0)
+		for i := 0; i < 3; i++ {
+			hostname := fmt.Sprintf("h%d", i)
+			host := setupNewHost(ctx, hostname, *cluster.ID)
+			hosts = append(hosts, host)
+		}
+		generateFullMeshConnectivity(ctx, "1.2.3.10", hosts...)
+		By("Approve Agents")
+		for _, host := range hosts {
+			hostkey := types.NamespacedName{
+				Namespace: Options.Namespace,
+				Name:      host.ID.String(),
+			}
+			Eventually(func() error {
+				agent := getAgentCRD(ctx, kubeClient, hostkey)
+				agent.Spec.Approved = true
+				return kubeClient.Update(ctx, agent)
+			}, "30s", "10s").Should(BeNil())
+		}
+
+		By("Wait for installing")
+		Eventually(func() string {
+			condition := FindStatusClusterDeploymentCondition(getClusterDeploymentCRD(ctx, kubeClient, clusterKey).Status.Conditions, hivev1.UnreachableCondition)
+			if condition != nil {
+				return condition.Message
+			}
+			return ""
+		}, "1m", "2s").Should(Equal(models.ClusterStatusInstalling))
+
+		By("Wait for finalizing")
+		for _, host := range hosts {
+			updateProgress(*host.ID, *cluster.ID, models.HostStageDone)
+		}
+
+		Eventually(func() string {
+			condition := FindStatusClusterDeploymentCondition(getClusterDeploymentCRD(ctx, kubeClient, clusterKey).Status.Conditions, hivev1.UnreachableCondition)
+			if condition != nil {
+				return condition.Message
+			}
+			return ""
+		}, "1m", "2s").Should(Equal(models.ClusterStatusFinalizing))
+
+		By("Complete Installation")
+		completeInstallation(agentBMClient, *cluster.ID)
+		isSuccess := true
+		_, err := agentBMClient.Installer.CompleteInstallation(ctx, &installer.CompleteInstallationParams{
+			ClusterID: *cluster.ID,
+			CompletionParams: &models.CompletionParams{
+				IsSuccess: &isSuccess,
+			},
+		})
+		Expect(err).NotTo(HaveOccurred())
+
+		By("Verify Day 2 Cluster")
 		Eventually(func() string {
 			condition := FindStatusClusterDeploymentCondition(getClusterDeploymentCRD(ctx, kubeClient, clusterKey).Status.Conditions, hivev1.UnreachableCondition)
 			if condition != nil {
 				return condition.Message
 			}
 			return ""
-		}, "1m", "2s").Should(Equal(models.ClusterStatusInstalled))
+		}, "2m", "2s").Should(Equal(models.ClusterStatusAddingHosts))
+		cluster = getClusterFromDB(ctx, kubeClient, db, clusterKey, waitForReconcileTimeout)
```

Comment:
shouldn't it be with eventually as well?

Reply:
I don't think so, as if the status has been updated to ClusterKindAddHostsCluster in the Eventually before, it means the Cluster has been already created in DB.